src/category_theory/quotient.lean

```diff
@@ -0,0 +1,104 @@
+/-
+Copyright (c) 2020 David Wärn. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Wärn
+-/
+
+import category_theory.functor
+import category_theory.eq_to_hom
+import group_theory.free_group
+
+/-!
+# Quotient category
+
+Constructs the quotient of a category by an arbitrary family of relations on its hom-sets,
+by introducing a type synonym for the objects, and identifying homs as necessary.
+
+This is analogous to 'the quotient of a group by the normal closure of a subset', rather
+than 'the quotient of a group by a normal subgroup'.
+-/
+
+universes v v₁ u u₁
+
+namespace category_theory
+
+variables {C : Type u} [category.{v} C]
+          (r : Π (a b : C), (a ⟶ b) → (a ⟶ b) → Prop)
+include r
+
+/-- A type synonom for `C`, thought of as the objects of the quotient category. -/
+structure quotient := (to_C : C)
+
+instance [inhabited C] : inhabited (quotient r) := ⟨⟨r, default C⟩⟩
+
+namespace quotient
+
+/-- Generates the closure of a family of relations w.r.t. composition from left and right. -/
+inductive comp_closure {s t : C} : (s ⟶ t) → (s ⟶ t) → Prop
```

Comment:
Should this use `{{s t : C}}` so that `comp_closure r` means the right thing?
(You can also use fancy braces by typing `\{}` instead of `{{`.)

Reply:
I think you're right! I had forgotten about these fancy braces.

Reply:
I've switched to fancy braces for `r`, `comp_closure r`, and `comp r`.

Reply:
(When review threads don't need require further response, it's easiest if the PR author marks them as resolved themselves. That way the reviewer, or other reviewers, don't have to re-read them when re-reviewing, only to find that everything is taken care of.)